docs/source/concepts/lifecycle.rst

```diff
@@ -1,22 +1,152 @@
+The simulation then enters its main processing section. The population
+is initialized, then the time step loop begins. Once the loop completes,
+results are calculated, gathered, and written out.
+
 .. contents::
    :depth: 2
    :local:
    :backlinks: none
 
-Outline
--------
-
-- Entry point variations (link to more detailed guide)
-- Initialization (create objects, compile configuration)
-- Setup (Wire components, load data)
-- Post setup
-- Population Initialization
-- Event loop
-  - time_step__prepare
-  - time_step
-  - time_step__cleanup
-  - collect_metrics
-- simulation_end
+Initialization
+--------------
+
+The initialization phase of ``vivarium`` lasts from when a user first interacts
+with a simulation :ref:`entry point <entry_points_concept>` to when the
+``__init__`` method of the :class:`vivarium.framework.engine.SimulationContext`
+completes.  In it, the following things happen (roughly in this order):
+
+1. Simulation outputs are configured. This is usually just setting up some
+   directories for the final results and the simulation logs.
+2. :term:`Model specification <Model Specification>` defaults are collected and
+   compiled along with the model specification file (if provided) into a
+   :class:`vivarium.config_tree.ConfigTree` object. Importantly,
+   default :term:`configuration <Configuration>` information from specific
+   components is not compiled into the :class:`vivarium.config_tree.ConfigTree`
+   at this point.
+3. A :class:`vivarium.framework.plugins.PluginManager` is generated around the
+   :term:`plugins <Plugin>` section of the model specification.  The plugin
+   manager is responsible for parsing the plugin section and instantiating
+   plugin controllers and interfaces for the framework.
+4. If :term:`components <Component>` are not provided directly by the user in
+   an interactive setting, the
+   :class:`vivarium.framework.components.ComponentConfigurationParser` is
+   created to parse and instantiate them from the
+   :term:`Model specification <Model Specification>`.
+5. The ``__init__`` method of the
+   :class:`vivarium.framework.engine.SimulationContext` is called with
+   the current :term:`configuration <Configuration>`, the instantiated
+   components, and the plugin manager.  This creates the
+   :ref:`builder <builder_concept>` which provides an interface to all the
+   simulation subsystems during the next phase of the simulation lifecycle.
+   It also registers all the simulation components with the
+   :class:`vivarium.framework.components.ComponentManager`.
+
+At this point, all input arguments have been parsed and all top-level
+components have been instantiated.  This is a useful phase in the simulation
+lifecycle because you can typically modify what components are in the system
+or how they are configured without any consequences.
+
+Setup
+-----
+
+In this stage, the framework moves to setting up the
+:term:`components <Component>`. For each top-level component, the framework
```

Comment:
I think top-level is not clear here to the uninitiated

Reply:
Will address in a different section and provide a reference here.